### PR TITLE
docs(igxTabs-group): Add api docs for igxTabs-group - with code snippets

### PR DESCRIPTION
**Pull request by alexalex972**

### CODE REVIEW

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 1]
```diff
@@ -24,50 +24,146 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+    * An @Input property that sets the value of the `label`.
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="folder">
+    *```
+    */
     @Input()
     public label: string;
 
+    /**
+    * An @Input property that sets the value of the `label`.
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: Sets the value of the icon.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 2]
```diff
@@ -24,50 +24,146 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+    public role = 'tabpanel';
+
+    /**
+     * An accessor that returns the class of the tab.
+     * ```html
+     * @ViewChild("MyTab")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabStyleClass = this.tab.styleClass;
+     *    this.cdr.detectChanges();
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: Do we need to put cdr.detectchanges() in the code snippets ? Maybe only if we are setting something.

Comment (alexalex972) [reply projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @hanastasov I'm not exactly sure, but for this particular snippet it wouldn't work without the cdr.detectchanges().

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 3]
```diff
@@ -24,50 +24,146 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+     * }
+     * ```
+     */
     get customTabTemplate(): TemplateRef<any> {
         return this._tabTemplate;
     }
 
+    /**
+     * An accessor that modifies the `_tabTemplate` property.
```
Comment (hanastasov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: Maybe it is better that we say the dewvelopers to use this to set a custom template to the tab. They do not need to know about the _tabTemplate property that will hold it.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts (hunk 2 above)
Comment (hanastasov) [reply projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: There is no change to detect here, so cdr.detectchanges() is excessive.

**Comment by kdinev** [comment]
@hanastasov I've actually requested in another PR, for `isDisabled` to be renamed to `disabled`.

**Comment by hanastasov** [comment]
@kdinev I don't quite get it, if we do it for the inputs, why would we wonder whether to rename it for the tab group as well ?

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 4]
```diff
@@ -77,17 +172,36 @@ export class IgxTabsGroupComponent implements AfterContentInit, AfterViewChecked
...
+     * @hidden
+     */
     public ngAfterViewChecked() {
         this._element.nativeElement.setAttribute('aria-labelledby', `igx-tab-item-${this.index}`);
         this._element.nativeElement.setAttribute('id', `igx-tabs__group-${this.index}`);
     }
 
+    /**
+     * A method that sets the focus on a tab.
+     * @memberOf {@link IgxDialogComponent}
```
Comment (sboykova) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: Wrong comment (IgxDialogComponent)

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 5]
```diff
@@ -77,17 +172,36 @@ export class IgxTabsGroupComponent implements AfterContentInit, AfterViewChecked
...
+     *```html
```
Comment (sboykova) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @ViewChild("MyTab")
    public tab: IgxTabsComponent;

    ngAfterViewInit(): void {
        // select second tab item
        this.tab.groups.toArray()[1].select();
    }

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 6]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+    * An @Input property that sets the value of the `label`.
+    *```typescript
+    *<igx-tabs-group label="Tab 1" icon="folder">
+    *```
+    */
     @Input()
     public label: string;
 
+    /**
+    * An @Input property that sets the value of the `icon`.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 Should this be a valid icon name from the icon set? Or is this a title of the icon? We should clarify.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 7]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+    *<igx-tabs-group label="Tab 1" icon="home">
+    *```
+    */
     @Input()
     public icon: string;
 
+    /**
+    * An @Input property that allows you to enable/disable the `IgxTabGroupComponent`.
+    *```html
+    *<igx-tabs-group label="Tab 2  Lorem ipsum dolor sit" icon="home" [isDisabled]="true">
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 `isDisabled` -> `disabled`.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 8]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
-    @HostBinding('attr.role') public role = 'tabpanel';
-
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.role')
+    public role = 'tabpanel';
+
+    /**
+     * An accessor that returns the class of the `IgxTabsGroupComponent`.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 "accessor" -> "property"

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 9]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+     * ```typescript
+     * @ViewChild("MyTabsGroup")
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 Maybe this should be hidden.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 10]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+     * }
+     * ```
+     */
     @HostBinding('class')
     get styleClass(): string {
         return 'igx-tabs__group';
     }
 
+    /**
+     * An accessor that returns the value of the `_itemStyle` property.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 Maybe this should be hidden or if it's not you should not mention private variables. Internal implementation is a subject of change.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 11]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+     *    let tabStyle = this.tab.itemStyle;
+     * }
+     * ```
+     */
+    public get itemStyle(): string {
+        return this._itemStyle;
+    }
+
+    /**
+     * An accessor that returns the `IgxTabItemComponent` component.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 Is this the active Tab item component? You should clarify.

File: projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts [hunk 12]
```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
...
+     * ```
+     */
     get relatedTab(): IgxTabItemComponent {
         if (this._tabs.tabs) {
             return this._tabs.tabs.toArray()[this.index];
         }
     }
 
+    /**
+     * An accessor that returns the value of the index of the `IgxTabsGroupComponent`.
```
Comment (mpavlinov) [inline projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts]: @alexalex972 Is this the index of the active tab? You should clarify.

**Pull request merged**